src/libYARP_dev/include/yarp/dev/IPidControlImpl.h

```diff
@@ -45,26 +45,32 @@ class YARP_dev_API yarp::dev::ImplementPidControl : public IPidControl
     */
     ImplementPidControl(yarp::dev::IPidControlRaw *y);
 
-    virtual bool setPid(int j, const Pid &pid);
-    virtual bool setPids(const Pid *pids);
-    virtual bool setReference(int j, double ref);
-    virtual bool setReferences(const double *refs);
-    virtual bool setErrorLimit(int j, double limit);
-    virtual bool setErrorLimits(const double *limits);
-    virtual bool getError(int j, double *err);
-    virtual bool getErrors(double *errs);
-    virtual bool getOutput(int j, double *out);
-    virtual bool getOutputs(double *outs);
-    virtual bool getPid(int j, Pid *pid);
-    virtual bool getPids(Pid *pids);
-    virtual bool getReference(int j, double *ref);
-    virtual bool getReferences(double *refs);
-    virtual bool getErrorLimit(int j, double *ref);
-    virtual bool getErrorLimits(double *refs);
-    virtual bool resetPid(int j);
-    virtual bool enablePid(int j);
-    virtual bool disablePid(int j);
-    virtual bool setOffset(int j, double v);
+    virtual bool setPid(const PidControlTypeEnum& pidtype, int j, const Pid &pid);
+    virtual bool setPids(const PidControlTypeEnum& pidtype, const Pid *pids);
+    virtual bool setPidReference(const PidControlTypeEnum& pidtype, int j, double ref);
+    virtual bool setPidReferences(const PidControlTypeEnum& pidtype, const double *refs);
+    virtual bool setPidErrorLimit(const PidControlTypeEnum& pidtype, int j, double limit);
+    virtual bool setPidErrorLimits(const PidControlTypeEnum& pidtype, const double *limits);
+    virtual bool getPidError(const PidControlTypeEnum& pidtype, int j, double *err);
```

Comment:
After the discussion f2f we decided to postpone this modification.
For the time being we simply do the transition to IPidControl